testutil/problem_stub.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// ProblemStub describes a subset of a Problem, and is used to verify
+// that problems look correct in tests.
+type ProblemStub struct {
+	Descriptor desc.Descriptor
```

Comment:
Problems don't always have a descriptor (at least not in the original design). The descriptor was actually just used as a convenience to specify the location. I think you should compare Location instead.

Reply:
I have used `Descriptor` for literally every rule I have written. I expect it to be by far the common case.

Reply:
I don't know where the old PackageLocation and SyntaxLocation methods went, but those don't have descriptors (unless this new library has them). We've written corp rules that lint both, so they won't be able to use this utility.

Reply:
I concur that location makes sense in those cases.

Reply:
You can probably still keep the SetDescriptor method, but it can then use the location to verify.

Reply:
So my goal here is to optimize for the common case. I am happy to add location if it makes sense later, but I still intend to leave this as-is for now.

Reply:
Fair enough. And although it's not completely relevant to this PR, can you point me to how rules will actually return a problem that points to package or syntax location? It will be relevant when we try to sync this.

Reply:
@robcapo for proto rules, returning the descriptor is required, as we use it to check if a rule is disabled or not by looking at its comments. We will need to find a way to enforce it. 

That said, rules involving syntax, package, imports or options are tricky, as they are not descriptors, but attributes belong to FileDescriptor. As a result, we need to 
1. create helpers to extract those location information from the original SourceCodeInfo, as we did before;
2. educate users that they can only disable those rules at the file comments.

Reply:
Until we address this we will be blocked from syncing. I believe we used to use file comments to determine if package or syntax rules were disabled.

Reply:
If >1 rules need syntax/package/options location, we will create a common help function, otherwise, I prefer that the rule itself deals with it, e.g., creating a `getSyntaxLocation(f FileDescriptor) *descriptor.SourceCodeInfo_Location`. I think it is not an issue, as @lukesneeringer already knows how to extract the original SourceCodeInfo, see [here](https://github.com/googleapis/api-linter/blob/9693cceee2f082cd894be60f10a88849e5ed3238/lint/rule.go#L287), but I am happy to help.

Reply:
I think @robcapo is right here. I originally changed `protoRule` to say (in the comments) "always use descriptor", but he makes a compelling case on package/syntax that I had not thought about. I went back and looked at my port of the syntax rule in AIP-191, and I used the file descriptor (meaning the error would be on line 1). That is not terrible, but it does not seem correct either.

I will address this in a separate PR. It is orthogonal to this one.

Reply:
Thanks Luke, and sorry to derail this PR with this conversation. I made #166 for the issue. Resolving this for now.